### PR TITLE
Add support for Catit Pixi Smart 6-meal Feeder

### LINKED ISSUE #614: Request support for Catit Pixi Smart 6-meal Feeder
# Log Message

```
2023-04-13 21:56:06.885 WARNING (MainThread) [custom_components.tuya_local.config_flow] Device matches None with quality of 0%. DPS: {"updated_at": 1681437364.7471426, "11": 0, "18": 1}
2023-04-13 21:56:06.887 WARNING (MainThread) [custom_components.tuya_local.config_flow] Report this to https://github.com/make-all/tuya-local/issues/
```

# Information about DPS mappings

```
{
  "result": {
    "category": "cwwsq",
    "functions": [
      {
        "code": "meal_plan",
        "dp_id": 1,
        "type": "Raw",
        "values": "{}"
      },
      {
        "code": "quick_feed",
        "dp_id": 2,
        "type": "Boolean",
        "values": "{}"
      },
      {
        "code": "factory_reset",
        "dp_id": 9,
        "type": "Boolean",
        "values": "{}"
      },
      {
        "code": "voice_times",
        "dp_id": 18,
        "type": "Integer",
        "values": "{\"unit\":\"\",\"min\":0,\"max\":1,\"scale\":0,\"step\":1}"
      }
    ],
    "status": [
      {
        "code": "meal_plan",
        "dp_id": 1,
        "type": "Raw",
        "values": "{}"
      },
      {
        "code": "quick_feed",
        "dp_id": 2,
        "type": "Boolean",
        "values": "{}"
      },
      {
        "code": "factory_reset",
        "dp_id": 9,
        "type": "Boolean",
        "values": "{}"
      },
      {
        "code": "battery_percentage",
        "dp_id": 11,
        "type": "Integer",
        "values": "{\"unit\":\"%\",\"min\":0,\"max\":100,\"scale\":0,\"step\":1}"
      },
      {
        "code": "feed_report",
        "dp_id": 15,
        "type": "Integer",
        "values": "{\"unit\":\"\",\"min\":0,\"max\":1000,\"scale\":0,\"step\":1}"
      },
      {
        "code": "voice_times",
        "dp_id": 18,
        "type": "Integer",
        "values": "{\"unit\":\"\",\"min\":0,\"max\":1,\"scale\":0,\"step\":1}"
      }
    ]
  },
  "success": true,
  "t": 1681522009383,
  "tid": "97fbb0b3db2c11edb3aad2d41d5e3422"
}
```

# Product ID

```
    "category": "cwwsq",
    "category_name": "Pet Feeder",
    "product_id": "wmkepdif6qzirl7a",
    "product_name": "Pixi 6-Meal feeder",
```

# Information about how the device functions

https://catit.ca/shop/pixi-smart-6-meal-feeder/
https://www.catit.com/help-advice/feeders-dishes/catit-pixi-smart-6-meal-feeder/

### PR DESCRIPTION
**Pull request by chemicalstorm**
Create new device configuration for the Catit Pixi Smart 6-meal Feeder.

Resolves #614

### CODE REVIEW

**Comment by chemicalstorm** [comment]
File content was heavily inspired by #585.

**Comment by chemicalstorm** [comment]
I'm trying to get a better understanding of 2 things:
- Not sure of the use of the `optional`. The device periodically send `Catit Pixi 6-Meal Feeder received {"11": 0, "15": 11, "18": 0, "full_poll": true}`, so I made the other optional but I don't think this is correct.
- Ideally, we should be able to map the voice setup to a simple boolean, but I need to read more on the `mapping` option.

**Comment by make-all** [comment]
The following mapping should be sufficient to turn the voice_setup input into a boolean for use in a switch.  The type should still be integer though, as that is the type of the underlying data.

```yaml
mapping:
  - dps_val: 0
    value: false
  - dps_val: 1
    value: true
```

File: custom_components/tuya_local/devices/catit_pixi_6meals_feeder.yaml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+      type: boolean
+      name: button
+      optional: True
+secondary_entities:
+  - entity: number
+    name: Meal plan
+    dps:
+    - id: 1
+      name: Meal Plan
+      # Meal plan is the following binary blob (1 byte per line), encoded in base64:
```
[the remark this answers is not in the document]
Comment (audricschiltknecht) [reply custom_components/tuya_local/devices/catit_pixi_6meals_feeder.yaml]: Thanks for the feedback!
I'm not sure I completely understood your comment, as I wasn't able to add the meal plan values as an additional attribute to other entities (I tried to put it on the "Feed report" one). However, I've changed the type to sensor, and not the value is displayed in the interface, even if the use/interaction is pretty limited as you mentioned.

[the remark this answers is not in the document]
Comment (chemicalstorm) [reply custom_components/tuya_local/devices/catit_pixi_6meals_feeder.yaml]: hanks for the feedback!
I'm not sure I completely understood your comment, as I wasn't able to add the meal plan values as an additional attribute to other entities (I tried to put it on the "Feed report" one). However, I've changed the type to sensor, and not the value is displayed in the interface, even if the use/interaction is pretty limited as you mentioned.

**Pull request merged**